libraries/AP_Scripting/applets/param-set.lua

```diff
@@ -0,0 +1,164 @@
+-- Inspect parameter sets received via MAVLink, determine action based
+-- on whitelist.
+
+-- When this script runs and ENABLE is true ArduPilot will stop
+--   processing parameter-sets via the GCS library.  Instead, this
+--   script becomes responsible for setting parameters, and it will
+--   only set parameters which are whitelisted.  Setting ENABLE to
+--   false will allow ArduPilot to set parameters normally.
+
+-- Setting SCR_ENABLE to false while this script is running in the
+--   ENABLE state is... not advised.
+
+-- How To Use
+-- 1. copy this script to the autopilot's "scripts" directory
+-- 2. within the "scripts" directory create a "modules" directory
+-- 3. copy the mavlink/mavlink_msgs_xxx files to the "scripts" directory in a "modules/mavlink" folder
+
+-- load mavlink message definitions from modules/mavlink directory
+local mavlink_msgs = require("mavlink/mavlink_msgs")
+
+-- global definitions
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+local UPDATE_INTERVAL_MS = 10           -- update at about 100hz
+
+-- get a reference to MAV_SYSID parameter so we can filter messages to
+--   just those parameter-sets aimed at this vehicle:
+MAV_SYSID = Parameter("MAV_SYSID")
+
+-- mavlink definitions
+local PARAM_SET_ID = 23
+local msg_map = {}
+msg_map[PARAM_SET_ID] = "PARAM_SET"
+
+local TEXT_PREFIX_STR = "param-set"    -- prefix for all text messages
+
+--
+-- parameter setup
+--
+local PARAM_TABLE_KEY = 92
+local PARAM_TABLE_PREFIX = "PS_"
```

Comment:
If we've got the space, how about we make "PS_" longer to add more help to the user?  Some ideas that fit within 16 characters:

- PARAM_SET_ENABLE
- PARMSET_ENABLE
- PSET_ENABLE

Reply:
> If we've got the space, how about we make "PS_" longer to add more help to the user? Some ideas that fit within 16 characters:
> 
>     * PARAM_SET_ENABLE

Yep, went with this one.